custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -560,13 +603,33 @@ def _predict_legacy_drum(self, data, model, **kwargs) -> RawPredictResponse:
             assert all(isinstance(label, str) for label in request_labels)
         extra_model_output = None
         if self._custom_hooks.get(CustomHooks.SCORE):
-            try:
-                # noinspection PyCallingNonCallable
-                predictions_df = self._custom_hooks.get(CustomHooks.SCORE)(data, model, **kwargs)
-            except Exception as exc:
-                self._log_and_raise_final_error(
-                    exc, "Model 'score' hook failed to make predictions."
-                )
+            if self._target_type == TargetType.TEXT_GENERATION and self._guard_pipeline:
```

Comment:
I think you cam move if clause inside try/except and have only one _log_and_raise_final_error